.nvidia-ci.yml

```diff
@@ -244,3 +244,69 @@ release:ngc-packaging:
   extends:
     - .dist-packaging
     - .release:ngc
+
+# Define the external image signing steps for NGC
+# Download the ngc cli binary for use in the sign steps
+.ngccli-setup:
+  before_script:
+    - export NGCLI_VERSION=3.41.4
+    - apt-get update && apt-get install -y curl unzip jq
+    - curl -sSLo ngccli_linux.zip https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions/${NGCLI_VERSION}/files/ngccli_linux.zip
+    - unzip ngccli_linux.zip
+    - chmod u+x ngc-cli/ngc
+    - export PATH=$(pwd)/ngc-cli:${PATH}
+    - NGC_CLI_API_KEY=${NGC_REGISTRY_TOKEN} ngc version upgrade --org nvidia -y
+
+# .sign forms the base of the deployment jobs which signs images in the CI registry.
+# This is extended with the image name and version to be deployed.
+.sign:ngc:
+  image: ubuntu:latest
+  stage: sign
+  variables:
+    IMAGE_NAME: "${OUT_IMAGE_NAME}"
+    IMAGE_TAG: "${OUT_IMAGE_TAG}"
+  retry:
+    max: 2
+  before_script:
+    - !reference [.ngccli-setup, before_script]
+    # We ensure that the OUT_IMAGE_TAG is set
+    - 'echo Image Tag: ${IMAGE_TAG} ; [[ -n "${IMAGE_TAG}" ]] || exit 1'
+  script:
+    - 'echo "Signing the image ${IMAGE_NAME}:${IMAGE_TAG}"'
+    - NGC_CLI_API_KEY="${NGC_REGISTRY_TOKEN}" ngc registry image publish --source ${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_NAME}:${IMAGE_TAG} --public --discoverable --allow-guest --sign --org nvidia
+
+sign:ngc-ubuntu20.04:
+  extends:
+    - .dist-ubuntu20.04
+    - .sign:ngc
+  needs:
+    - release:ngc-ubuntu20.04
+  rules:
+    - if: $CI_COMMIT_TAG
+  variables:
+    OUT_IMAGE_TAG: "${CI_COMMIT_TAG}-${DIST}"
+    OUT_IMAGE_NAME: "${NGC_REGISTRY_IMAGE}"
```

Comment:
The `rules` and `variables` can be specified ONCE in `.sign:ngc` instead of for each job.

Reply:
done